src/bootstrap/setup.rs

```diff
@@ -104,19 +126,35 @@ pub fn setup(src_path: &Path, profile: Profile) {
 
 // Used to get the path for `Subcommand::Setup`
 pub fn interactive_path() -> io::Result<Profile> {
-    let mut input = String::new();
-    println!(
-        "Welcome to the Rust project! What do you want to do with x.py?
-a) Contribute to the standard library
-b) Contribute to the compiler
-c) Contribute to the compiler, and also modify LLVM or codegen
-d) Install Rust from source"
-    );
+    println!("Welcome to the Rust project! What do you want to do with x.py?");
+
+    fn abbrev_all() -> impl Iterator<Item = (String, Profile)> {
```

Comment:
Why not just store this as `Vec` rather than creating it so many times?

Reply:
Performance does not matter for bootstrap. It should be correct and easy to write and read, everything else is secondary.

Reply:
Yes, I don't mean for performance. Handling vec is indeed easier than iterator.

Reply:
I think you are suggesting roughly `let abbrev_all : Vec<(String,Profile)> = ('a'..).map(...).collect()`.  I don't think there is much in it.  There are three uses, all of which are using the result as an iterator.  One is calling `.map` and would need to have `.iter()` added.  Iterators seem very idiomatic style in Rust.  So I prefer it this way.

The main possible benefit of a variable (or a closure) would be ability to omit the types.  I did consider changing the fn to a closure when it became a local, here.  But I think specifying the entry type is useful documentation.